reference/shaders-hlsl/comp/ssbo-array.comp

```diff
@@ -0,0 +1,25 @@
+struct SSBO
+{
+    float4 data[1];
+};
+
+RWStructuredBuffer<SSBO> ssbos : register(u0);
+
+static uint3 gl_GlobalInvocationID;
+struct SPIRV_Cross_Input
+{
+    uint3 gl_GlobalInvocationID : SV_DispatchThreadID;
+};
+
+void comp_main()
+{
+    uint ident = gl_GlobalInvocationID.x;
+    ssbos[1].data[ident] = ssbos[0].data[ident];
```

Comment:
You're right this test case is incorrect. The way this is working right now is that it doesn't support an array of separate SSBOs.
```
layout(std430, binding = 0) buffer SSBO
{
    vec4 data[];
} ssbos[2];
```
is being interpreted as meaning "I have **one** SSBO which **is** and array of `struct { vec4 data[]; }`," instead of "I have an **array of** SSBOs each which contains `vec4 data[];`.

So I think the tough point here is that the contents of the SSBOs are unsized arrays.
Each UAV binding becomes an `RWStructuredBuffer<T>` which is fine if `T` has a fixed size.
This however, doesn't work so well if T has an unsized array. `RWStructuredBuffer<T> buffer` is sort of implicitly the same as `T buffer[]`.

I'm not sure why a `[1]` was added to the generated code, but the generated SPIRV should be:
```
struct SSBO
{
    float4 data[];
};

RWStructuredBuffer<SSBO> ssbos : register(u0);
```
which doesn't really make sense because then `RWStructuredBuffer<SSBO> ssbos` which is implicitly `SSBO ssbos[]` is an unsized array containing unsized float4 arrays.

If, however, `T` is fixed size, then there are no issues. For example:
```
struct Particle {
    vec2 pos;
    vec2 vel;
};

layout(std430, set = 0, binding = 0) buffer ParticlesA {
    Particle particle;
} particlesA[1000];
```
becomes something like:
```
struct Particle {
    vec2 pos;
    vec2 vel;
};

struct _ParticlesA {
    Particle particle;
};

RWStructuredBuffer<_ParticlesA> particlesA: register(u0);
```